tests/integration/servicecatalog/service_test.go

```diff
@@ -104,17 +105,60 @@ var _ = Describe("odoServiceE2e", func() {
 
 	Context("When working from outside a component dir", func() {
 		JustBeforeEach(func() {
-			project = helper.CreateRandProject()
 			context = helper.CreateNewContext()
+			os.Setenv("GLOBALODOCONFIG", filepath.Join(context, "config.yaml"))
+			project = helper.CreateRandProject()
 			app = helper.RandString(7)
+			serviceName = "odo-postgres-service"
 			originalDir = helper.Getwd()
 			helper.Chdir(context)
+			SetDefaultConsistentlyDuration(30 * time.Second)
 		})
 		JustAfterEach(func() {
 			helper.DeleteProject(project)
 			helper.DeleteDir(context)
 			helper.Chdir(originalDir)
+			os.Unsetenv("GLOBALODOCONFIG")
+		})
+
+		It("should be able to create, list and delete a service using a given value for --context", func() {
+			// create a component by copying the example
+			helper.CopyExample(filepath.Join("source", "python"), context)
+			helper.CmdShouldPass("odo", "create", "python", "--app", app, "--project", project)
+
+			// cd to the originalDir to create service using --context
+			helper.Chdir(originalDir)
+			helper.CmdShouldPass("odo", "service", "create", "dh-postgresql-apb", "--plan", "dev",
+				"-p", "postgresql_user=luke", "-p", "postgresql_password=secret",
+				"-p", "postgresql_database=my_data", "-p", "postgresql_version=9.6", serviceName,
+				"--context", context,
+			)
+
+			// now check if listing the service using --context works
+			ocArgs := []string{"get", "serviceinstance", "-o", "name", "-n", project}
+			helper.WaitForCmdOut("oc", ocArgs, 1, true, func(output string) bool {
+				return strings.Contains(output, serviceName)
+			})
+			stdOut := helper.CmdShouldPass("odo", "service", "list", "--context", context)
+			Expect(stdOut).To(ContainSubstring(serviceName))
+
+			// now check if deleting the service using --context works
+			stdOut = helper.CmdShouldPass("odo", "service", "delete", "-f", serviceName, "--context", context)
+			Expect(stdOut).To(ContainSubstring(serviceName))
+		})
+
+		It("should fail to create a service when given directory doesn't contain a context", func() {
+			// cd to the originalDir to create service using --context
+			helper.Chdir(originalDir)
+			stdOut := helper.CmdShouldFail("odo", "service", "create", "dh-postgresql-apb", "--plan", "dev",
```

Comment:
Hmm, it seems the command returns exit code ```0```. So the command actually does not fail.

@dharmit use ```CmdShouldPass``` instead.

Reply:
I think the right thing to do would be remove the `"--context", context,` appearing a couple of lines below the one you pointed because as we can see from the `It(should fail to create a service when given directory doesn't contain a context")`, this test block is supposed to fail when working without a context.

Reply:
Right, Have you tested it locally, i mean the same steps in the terminal ?

I doubt it will give a none zero exit code. Anyway lets see

Reply:
> I think the right thing to do would be remove the "--context", context,

In this scenario there is no component in the context dir. So i feel even after removing the context flag won't work as per the expectation.

Reply:
I've not checked tests locally for this instance. For some reason the tests were not getting started and I took the lazy approach of pushing it to a PR. Let me check this locally because what you say about removing the context flag also makes sense.

Reply:
#2051 is a result of this discussion. Marking this conversation resolved.